src/mgr/TTLCache.h

```diff
@@ -93,29 +95,23 @@ void TTLCache<Key, Value>::insert(Key key, Value value) {
 }
 
 template <class Key, class Value>
-void TTLCache<Key, Value>::update_ttl() {
-	ttl = g_conf().get_val<int64_t>("mgr_ttl_cache_expire_seconds");
-}
-
-template <class Key, class Value> Value TTLCache<Key, Value>::get(Key key) {
+boost::optional<Value> TTLCache<Key, Value>::get(Key key) {
+  boost::optional<Value> value;
   if (!this->exists(key)) {
-    return nullptr;
+    return value;
   }
   if (expired(key)) {
-             << Py_REFCNT(get_value(key)) << "\n";
     erase(key);
-    return nullptr;
-  }
-  if (PyObject *obj = dynamic_cast<PyObject *>(get_value(key))) {
-    Py_INCREF(obj);
+    return value;
   }
   this->Cache<Key, value_type>::hit();
-  return get_value(key);
+  value = {get_value(key)};
+  return value;
 }
 
 template <class Key, class Value> void TTLCache<Key, Value>::erase(Key key) {
-  if (PyObject *obj = dynamic_cast<PyObject *>(get_value(key))) {
-    Py_DECREF(obj);
+  if (erase_callback) {
```

Comment:
While we don't change storing pyobjects I need a way to know when the PyObject is erased from the cache so I know when to decrease the ref counter.